src/libraries/System.Drawing.Common/src/System/Drawing/Printing/PrinterSettings.cs

```diff
@@ -1042,17 +1043,20 @@ internal PaperSize[] Get_PaperSizes()
             FastDeviceCapabilities(SafeNativeMethods.DC_PAPERSIZE, dimensionsBuffer, -1, printerName);
 
             PaperSize[] result = new PaperSize[count];
+            byte* pNamesBuffer = (byte*)namesBuffer;
+            byte* pKindsBuffer = (byte*)kindsBuffer;
+            byte* pDimensionsBuffer = (byte*)dimensionsBuffer;
             for (int i = 0; i < count; i++)
             {
-                string name = Marshal.PtrToStringAuto((IntPtr)(checked((long)namesBuffer + stringSize * i)), 64)!;
+                string name = Marshal.PtrToStringAuto((nint)(pNamesBuffer + stringSize * i), 64)!;
```

Comment:
```suggestion
                string name = Marshal.PtrToStringAuto((nint)(pNamesBuffer + stringSize * (nint)i), 64)!;
```
The "manual" indexing like this one should be using `nint` to compute the offset from the pointer to avoid risk of integer overflows. The easiest way to achieve that is to cast the unscaled index to `nint` that will cause the multiplication to be performed as `nint`.

(Similar pre-existing problem is in number of places in the code touched in this PR. Search for `* i` or `i *`. I will leave it up to your whether to fix them.)

Reply:
> I will leave it up to your whether to fix them.

I should fix, do not want to cause any issue 😛 

Question: why don't declare the `i` with `nint`? I assume it is only apply for int pointers, doesn't apply for example to the short pointer `pKindsBuffer`  below

Reply:
> why don't declare the i with nint?

You can try that, but expect that it will cause other problems. 

For clarity, it may be better to have `nint` cast in the expression that computes the pointer. It makes the intent more obvious.

If you change `i` to `nint` and the loop is more than a few lines, it is not immediately obvious why it needs to be `nint`. Somebody may change it back to `int`. 

Right, it only applies to places that follow the `pFoo + i * sizeOfElement` pattern.